controller/controller-http/src/main/java/com/pinterest/rocksplicator/controller/resource/Clusters.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pinterest.rocksplicator.controller.resource;
+
+import com.pinterest.rocksplicator.controller.bean.TaskBean;
+
+import java.util.List;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * @author Ang Xu (angxu@pinterest.com)
+ */
+@Path("/v1/clusters/{clusterName : [a-zA-Z0-9\\-_]+}")
+public class Clusters {
+
+  /**
+   * Retrieves all tasks belong to a given cluster by cluster name.
+   *
+   * @param clusterName name of the cluster
```

Comment:
add an optional state parameter? if it's not set, we retrive all, otherwise we retrieve the ones in the specified state. and rename the function to getTasks.